src/bugsnag_tools/core.clj

```diff
@@ -15,14 +15,38 @@
     path
     (str host path)))
 
+(defn retry
+  "Given a function, f, run it after delay-ms milliseconds have
+  passed."
+  [f delay-ms]
+  (Thread/sleep delay-ms)
+  (f))
+
+(defn rate-limit-middleware
+  [client]
+  (fn [req]
+    (try
+      (client req)
+      (catch clojure.lang.ExceptionInfo e
+        (let [data (ex-data e)]
+          (if (= (:status data) 429)
+            (do
+              (println "limiting...")
+              (retry #(client req)
+                    (-> (get-in data [:headers "Retry-After"] "10")
```

Comment:
Wow, that's a really long delay, I assumed the rate limiting would have been on the order of milliseconds.  I think 1 second would be just fine.